### PR TITLE
A few of the tests that are still missing

### PR DESCRIPTION
**Pull request by christophfink**

### CODE REVIEW

**Comment by christophfink** [comment]
On top of providing many of the still missing tests, this pull request also changed config.py and classpath.py. In particular, the core functionality of these two modules are wrapped in functions or classes. This was necessary to allow tests (that necessarily have to repeat what was run, already).

File: src/r5py/r5/transport_network.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def from_directory(cls, path):
                     ),
                     RuntimeWarning,
                 )
-        except KeyError:
+        except IndexError:
```
Comment (wklumpen) [inline src/r5py/r5/transport_network.py:102]: What prompted switching from `KeyError` to `IndexError`? Should we include both?

**Review by wklumpen — APPROVED** [review]
Left one minor query but overall tests look good.

File: src/r5py/r5/transport_network.py (hunk 1 above)
Comment (christophfink) [reply src/r5py/r5/transport_network.py:102]: It was wrong :D and the test showed that. Hooray for test-driven development

**Comment by christophfink** [comment]
For the record: the failing test is a GitHub web server error, it returns http error code 503 (server error) when fetching the r5 jar (which works in all the remaining tests).

**Pull request merged**